### PR TITLE
Implement asAction

### LINKED ISSUE #5: Proposal: Relay.asAction() method to match asObservable()
There are times where I only want to expose the action. I could cast or wrap in a method that returns an Action1, but it might be nice to just have an `asAction()` method for convenience on the base `Relay` object as well that does this.

### PR DESCRIPTION
**Pull request by ZacSweers**
Resolves #5

### CODE REVIEW

File: src/test/java/com/jakewharton/rxrelay/AsActionTest.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        Observer<Object> o = mock(Observer.class);
+
+        src.subscribe(o);
+
+        action.call(1);
+
+        verify(o).onNext(1);
+        verify(o, never()).onCompleted();
+        verify(o, never()).onError(any(Throwable.class));
+    }
```
Comment (JakeWharton) [inline src/test/java/com/jakewharton/rxrelay/AsActionTest.java]: two-space indent whole file

Comment (ZacSweers) [inline src/test/java/com/jakewharton/rxrelay/AsActionTest.java]: Whoops 8c5278e

File: src/test/java/com/jakewharton/rxrelay/AsActionTest.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  @Test
+  public void testHiding() {
+   PublishRelay<Integer> src = PublishRelay.create();
+
+   Action1<Integer> action = src.asAction();
+
+   assertFalse(action instanceof PublishRelay);
+
+   @SuppressWarnings("unchecked")
+   Observer<Object> o = mock(Observer.class);
```
Comment (JakeWharton) [inline src/test/java/com/jakewharton/rxrelay/AsActionTest.java]: You can use `TestSubscriber` for this and avoid mocking.

File: src/test/java/com/jakewharton/rxrelay/AsActionTest.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+import static org.junit.Assert.assertFalse;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+
+public final class AsActionTest {
+
+  @Test
+  public void testHiding() {
```
Comment (JakeWharton) [inline src/test/java/com/jakewharton/rxrelay/AsActionTest.java]: no need for test prefix

Comment (ZacSweers) [inline src/test/java/com/jakewharton/rxrelay/AsActionTest.java]: 357344e

File: src/test/java/com/jakewharton/rxrelay/AsActionTest.java (hunk 2 above)
Comment (ZacSweers) [inline src/test/java/com/jakewharton/rxrelay/AsActionTest.java]: 357344e

File: src/main/java/com/jakewharton/rxrelay/Relay.java [hunk 4]
```diff
@@ -26,6 +26,22 @@ protected Relay(OnSubscribe<R> onSubscribe) {
   }
 
   /**
+   * Portrays a object of an Action1 implementation as a simple Relay object. This is useful, for instance,
```
Comment (JakeWharton) [inline src/main/java/com/jakewharton/rxrelay/Relay.java]: Is this copied from `asObservable()`? The first sentence reads really weird, almost backwards.

Comment (ZacSweers) [inline src/main/java/com/jakewharton/rxrelay/Relay.java]: Hmm, yeah you're right. Was trying to keep the language consistent but it's definitely backward here. I'll fix it

Comment (ZacSweers) [inline src/main/java/com/jakewharton/rxrelay/Relay.java]: Done a57f322

**Pull request merged**